### DISCUSSION TITLE
How to set a default addition to generated pyproject.toml?

### DISCUSSION THREAD

**Discussion by BruceEckel** [astral-sh/rye#398] [General]
I would like, in every `pyproject.toml` I create using rye, to automatically include the following lines when rye creates the file:

```
[tool.black]
line-length = 47
```

I imagine this would be part of the config file `config.toml` in the `.rye` folder, but I don't see anything about how to do it in https://rye-up.com/guide/config/.

**Comment by cnpryer** [comment]
This gives me plugin-territory vibes, but maybe others disagree. Would make sense as a scaffolding plugin, assuming a future plugin system is attainable.

**Reply by nathanjmcdougall** [reply]
https://rye-up.com/philosophy/#opinionated-defaults

> Rye shall ship with the one true formatter

I think it's a question of whether formatters and linters should be given first-class status, or whether they should be plugins.

**Reply by cnpryer** [reply]
Yea the way I'd probably do it is ship `ruff` with the `rye` toolchain when it's ready and shell out to installed alternatives like `black` otherwise.

Here I was mainly referring to defaults for configuration.
